lib/jira/client/axios.js

```diff
@@ -8,80 +8,139 @@ const JiraClientError = require('./jira-client-error');
+ */
 function getAuthMiddleware(secret) {
-  return (config) => {
-    const { query, pathname } = url.parse(config.url, true);
-
-    const jwtToken = jwt.encode({
-      ...getExpirationInSeconds(),
-      iss,
-      qsh: jwt.createQueryStringHash({
-        method: config.method,
-        originalUrl: pathname,
-        query,
-      }),
-    }, secret);
-
-    return {
-      ...config,
-      headers: {
-        ...config.headers,
-        Authorization: `JWT ${jwtToken}`,
-      },
-    };
-  };
+  return (
+    /**
+     * @param {import('axios').AxiosRequestConfig} config - The config for the outgoing request.
+     * @returns {import('axios').AxiosRequestConfig} Updated axios config with authentication token.
+     */
+    (config) => {
+      const { query, pathname } = url.parse(config.url, true);
+
+      const jwtToken = jwt.encode({
+        ...getExpirationInSeconds(),
+        iss,
+        qsh: jwt.createQueryStringHash({
+          method: config.method,
+          originalUrl: pathname,
+          query,
+        }),
+      }, secret);
+
+      return {
+        ...config,
+        headers: {
+          ...config.headers,
+          Authorization: `JWT ${jwtToken}`,
+        },
+      };
+    });
 }
 
-function getErrorMiddleware(logger) {
-  return (error) => {
-    if (error.response) {
-      const { status, statusText } = error.response || {};
-      logger.debug({
-        params: error.config.fields,
-      }, `Jira request: ${error.request.method} ${error.request.path} - ${status} ${statusText}`);
+/**
+ * Mapping of Jira Dev Error Codes to nice strings.
+ *
+ * See {@link https://developer.atlassian.com/cloud/jira/software/rest/#api-devinfo-0-10-bulk-post} for details on the API errors.
+ */
+const JiraErrorCodes = {
+  400: 'Request had incorrect format.',
+  401: 'Missing a JWT token, or token is invalid.',
+  403: 'The JWT token used does not correspond to an app that defines the jiraDevelopmentTool module, or the app does not define the \'WRITE\' scope',
+  413: 'Data is too large. Submit fewer devinfo entities in each payload.',
+  429: 'API rate limit has been exceeded.',
+};
 
-      return Promise.reject(new JiraClientError(error));
-    } else {
-      return Promise.reject(error);
-    }
-  };
+/**
+ * Middleware to enhance failed requests in Jira.
+ *
+ * @param {import('probot').Logger} logger - The probot logger instance
+ */
+function getErrorMiddleware(logger) {
+  return (
+    /**
+     * Potentially enrich the promise's rejection.
+     *
+     * @param {import('axios').AxiosError} error - The error response from Axios
+     * @returns {Promise<Error>} The rejected promise
+     */
+    (error) => {
+      if (error.response) {
+        const { status, statusText } = error.response || {};
+        logger.debug({
+          params: error.config.fields,
+        }, `Jira request: ${error.request.method} ${error.request.path} - ${status} ${statusText}`);
+
+        if (status in JiraErrorCodes) {
```

Comment:
whats the diff between status and error.response.status?

Reply:
`status` actually comes from `error.response` on line 73 above! Same thing! I just forgot to use it further down.